src/utils/unique_email_enforcement/storage.ts

```diff
@@ -0,0 +1,77 @@
+import * as E from "fp-ts/lib/Either";
+import * as t from "io-ts";
+
+import { TableClient, odata, TableEntityResult } from "@azure/data-tables";
+import { EmailString } from "@pagopa/ts-commons/lib/strings";
+
+import { ProfileEmail, ProfileEmailReader, ProfileEmailWriter } from "./index";
+
+// Generates AsyncIterable<ProfileEmail> from AsyncIterable<TableEntityResult>
+export async function* toProfileEmailsAsyncIterator(
+  iterator: AsyncIterableIterator<TableEntityResult<unknown>>
+): AsyncIterableIterator<ProfileEmail> {
+  for await (const item of iterator) {
+    const profileEmail = ProfileEmail.decode({
+      email: item.partitionKey,
+      fiscalCode: item.rowKey
+    });
+    if (E.isRight(profileEmail)) {
+      yield profileEmail.right;
+    }
+  }
+}
+
+const TableEntity = t.type({
+  partitionKey: t.string,
+  rowKey: t.string
+});
+
+type TableEntity = t.TypeOf<typeof TableEntity>;
+
+const ProfileEmailToTableEntity = new t.Type<ProfileEmail, TableEntity>(
+  "TableEntityFromProfileEmail",
+  ProfileEmail.is,
+  ProfileEmail.decode,
+  ({ email, fiscalCode }) => ({
+    partitionKey: email.toLowerCase(),
+    rowKey: fiscalCode
+  })
+);
```

Comment:
We can take advantage of this codec in `toProfileEmailsAsyncIterator` by updating it this way

```typescript
const ProfileEmailToTableEntity = new t.Type<ProfileEmail, TableEntity>(
  "TableEntityFromProfileEmail",
  ProfileEmail.is,
  _ =>
    pipe(
      TableEntity.decode(_),
      E.chain(item =>
        ProfileEmail.decode({
          email: item.partitionKey,
          fiscalCode: item.rowKey
        })
      )
    ),
  ({ email, fiscalCode }) => ({
    partitionKey: email.toLowerCase(),
    rowKey: fiscalCode
  })
);
```

Reply:
done!